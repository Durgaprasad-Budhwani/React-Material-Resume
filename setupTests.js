/**
 * Created by Durgaprasad Budhwani on 12/14/2017.
 */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Element.prototype.getBoundingClientRect = function(){
    return {
        y: -100,
        top: -100
    }
};

configure({ adapter: new Adapter() });