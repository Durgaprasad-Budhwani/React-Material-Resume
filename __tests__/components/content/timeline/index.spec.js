/**
 * Created by durgaprasad on 10/10/17.
 */
/* global expect,it,describe,afterEach */
import React from 'react';
import ExperienceContainer, {Experiences} from '../../../../src/components/content/timeline';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import '../../../../setupTests';

describe('<Experiences />', () => {
    const classes = {
        container: {},
        barClass: {},
        relative: {},
        clear: {}
    };
    const experience = {
        year: "year",
        company: 'company',
        title: 'title',
        description: 'description',
    };
    afterEach(() => {
        fetchMock.restore();
    });
    
    it('renders <ExperienceContainer />', () => {
        let experiences = [ experience ];
        fetchMock.mock('experience.json', experiences);
        const tree = renderer.create(
            <ExperienceContainer
                classes={classes}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
    
    it('renders <Experiences />', () => {
        let experiences = [ experience ];
        fetchMock.mock('experience.json', experiences);
        const wrapper = mount(
            <Experiences
                classes={classes}
            />
        );
        let instance = wrapper.instance();
        let experienceWrapper = shallow(instance._renderExperience(experience, 1));
        expect(toJson(experienceWrapper)).toMatchSnapshot();
    });
});

