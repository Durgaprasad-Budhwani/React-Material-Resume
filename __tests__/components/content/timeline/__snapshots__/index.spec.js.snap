// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders timeline 1`] = `
<div
  className="MuiGrid-typeContainer-5 MuiGrid-typeItem-6 MuiGrid-direction-xs-column-7 MuiGrid-justify-xs-center-21 MuiGrid-grid-xs-12-41 Timeline-container-1"
>
  <span
    className="SectionTitle-sectionTitle-94"
  >
    Work Experience
  </span>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="Timeline-clear-4"
    >
      <div
        className="TimelineBar-bar-95 Timeline-barClass-2"
      />
      <div>
        <div
          className="EventContainer-bigScreenBox-101 EventContainer-boxRight-103"
        >
          <div
            className="animated"
            style={
              Object {
                "animationDuration": "0s",
                "visibility": "hidden",
              }
            }
          >
            <div
              className="MuiPaper-root-110 MuiPaper-shadow2-114 MuiPaper-rounded-111 EventContainer-paper-97 EventContainer-paperRight-99"
            >
              <span
                className="EventContainer-arrow-105 EventContainer-arrowLeft-107"
              />
              <div
                align="center"
                className="MuiGrid-typeContainer-5 MuiGrid-direction-xs-column-7"
              >
                <div
                  className="Event-date-137"
                >
                  2014 - Present
                </div>
                <h3
                  className="Event-header3-138"
                >
                  Varian Medical System
                </h3>
                <h4
                  className="Event-header4-139"
                >
                  Full Stack Developer
                </h4>
                <p
                  className="Event-description-140"
                >
                  TBD TBD TBD TBD TBD TBD
                </p>
              </div>
            </div>
          </div>
          <span
            className="EventContainer-dot-104 EventContainer-dotLeft-109"
          />
        </div>
        <div
          className="MuiPaper-root-110 MuiPaper-shadow2-114 MuiPaper-rounded-111 EventContainer-smallScreenBox-96"
        >
          <div
            align="center"
            className="MuiGrid-typeContainer-5 MuiGrid-direction-xs-column-7"
          >
            <div
              className="Event-date-137"
            >
              2014 - Present
            </div>
            <h3
              className="Event-header3-138"
            >
              Varian Medical System
            </h3>
            <h4
              className="Event-header4-139"
            >
              Full Stack Developer
            </h4>
            <p
              className="Event-description-140"
            >
              TBD TBD TBD TBD TBD TBD
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          className="EventContainer-bigScreenBox-101 EventContainer-boxLeft-102"
        >
          <div
            className="animated"
            style={
              Object {
                "animationDuration": "0s",
                "visibility": "hidden",
              }
            }
          >
            <div
              className="MuiPaper-root-110 MuiPaper-shadow2-114 MuiPaper-rounded-111 EventContainer-paper-97 EventContainer-paperLeft-98"
            >
              <span
                className="EventContainer-arrow-105 EventContainer-arrowRight-106"
              />
              <div
                align="center"
                className="MuiGrid-typeContainer-5 MuiGrid-direction-xs-column-7"
              >
                <div
                  className="Event-date-137"
                >
                  2014 - Present
                </div>
                <h3
                  className="Event-header3-138"
                >
                  Varian Medical System
                </h3>
                <h4
                  className="Event-header4-139"
                >
                  Full Stack Developer
                </h4>
                <p
                  className="Event-description-140"
                >
                  TBD
                </p>
              </div>
            </div>
          </div>
          <span
            className="EventContainer-dot-104 EventContainer-dotRight-108"
          />
        </div>
        <div
          className="MuiPaper-root-110 MuiPaper-shadow2-114 MuiPaper-rounded-111 EventContainer-smallScreenBox-96"
        >
          <div
            align="center"
            className="MuiGrid-typeContainer-5 MuiGrid-direction-xs-column-7"
          >
            <div
              className="Event-date-137"
            >
              2014 - Present
            </div>
            <h3
              className="Event-header3-138"
            >
              Varian Medical System
            </h3>
            <h4
              className="Event-header4-139"
            >
              Full Stack Developer
            </h4>
            <p
              className="Event-description-140"
            >
              TBD
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
